src/WebJobs.Script.WebHost/Controllers/AdminController.cs

```diff
@@ -148,6 +148,35 @@ public IHttpActionResult Ping()
         }
 
         [HttpPost]
+        [Route("admin/host/log")]
+        public IHttpActionResult Log(IEnumerable<HostLogEntry> logEntries)
+        {
+            foreach (var logEntry in logEntries)
+            {
+                var traceEvent = new TraceEvent(logEntry.Level, logEntry.Message, logEntry.Source);
+                _traceWriter.Trace(traceEvent);
+
+                switch (logEntry.Level)
+                {
+                    case TraceLevel.Verbose:
```

Comment:
@brettsam what is the pattern we can use here so we take advantage of structured logging? It would be good to make sure App Insights understands the different properties of this object.